modules/auxiliary/gather/ldap_query.rb

```diff
@@ -67,75 +67,75 @@ def initialize(info = {})
   end
 
   def initialize_actions
-    filename = 'ldap_queries_default.yaml'
-    @user_config_file_path = File.join(::Msf::Config.get_config_root.to_s, filename)
-    default_config_file_path = File.join(::Msf::Config.data_directory, 'auxiliary', 'gather', 'ldap_query', filename)
+    user_config_file_path = File.join(::Msf::Config.get_config_root, 'ldap_queries.yaml')
+    default_config_file_path = File.join(::Msf::Config.data_directory, 'auxiliary', 'gather', 'ldap_query', 'ldap_queries_default.yaml')
 
-    unless File.exist?(@user_config_file_path)
+    @loaded_queries = safe_load_queries(default_config_file_path) || []
+    if File.exist?(user_config_file_path)
+      @loaded_queries.concat(safe_load_queries(user_config_file_path))
+    else
       # If the user config file doesn't exist, then initialize it with a sample entry.
       # Users can adjust this file to overwrite default actions to retrieve different attributes etc by default.
-      content = "---
-queries:
-  - action: SAMPLE_ACTION
-    description: 'A description.'
-    filter: '(objectClass=*)'
-    attributes:
-      - dn
-      - objectClass"
-      File.write(@user_config_file_path, content)
-    end
-
-    begin
-      @default_settings_file_path = @user_config_file_path
-      @default_settings = YAML.safe_load(File.binread(@default_settings_file_path))
-    rescue StandardError => e
-      fail_with(Failure::BadConfig, "Couldn't parse #{@default_settings_file_path}, error was: #{e}")
-    end
-
-    begin
-      user_settings = YAML.safe_load(File.binread(default_config_file_path))
-    rescue StandardError => e
-      fail_with(Failure::BadConfig, "Couldn't parse #{default_config_file_path}, error was: #{e}")
-    end
-
-    unless @default_settings['queries']&.class == Array && user_settings['queries']&.class == Array && !@default_settings['queries'].empty? && !user_settings['queries'].empty?
-      fail_with(Failure::BadConfig, "Either #{@default_settings_file_path} did not contain any queries or #{@user_config_file_path} did not contain any queries!")
+      template = File.join(::Msf::Config.data_directory, 'auxiliary', 'gather', 'ldap_query', 'ldap_queries_template.yaml')
+      FileUtils.cp(template, user_config_file_path) if File.exist?(template)
     end
 
     # Combine the user settings with the default settings and then uniq them such that we only have one copy
     # of each ACTION, however we use the user's custom settings if they have tweaked anything to prevent overriding
     # their custom adjustments.
-    @default_settings['queries'].concat(user_settings['queries'])
-    @default_settings['queries'].uniq! do |q|
-      q['action']
-    end
+    @loaded_queries = Hash[*@loaded_queries.map { |h| [h['action'], h] }.flatten]
```

Comment:
```suggestion
    @loaded_queries = @loaded_queries.map { |h| [h['action'], h] }.to_h
```

Taken after discussions on Slack about how the code above worked. Really either one works fine this one above just read easier and seems to follow existing code patterns a bit more.